core/gas/service.go

```diff
@@ -38,6 +38,7 @@ type Meter interface {
 	Consume(amount Gas, descriptor string) error
 	Refund(amount Gas, descriptor string) error
 	Remaining() Gas
+	Consumed() Gas
```

Comment:
should avoid adding this

Reply:
i thinks return remaining in consumed is not corect since remaining implied `limit - consumed ` but we need to get the consumed amount only

Reply:
https://github.com/cosmos/cosmos-sdk/issues/21440 & https://github.com/cosmos/cosmos-sdk/issues/21440#issuecomment-2315159000 :D

Reply:
It's in stf implementation too https://github.com/cosmos/cosmos-sdk/blob/355f4d7e422f111b6a61243872adc6d5661f4cad/server/v2/stf/gas/meter.go#L24, so i'd say it is great to add indeed and fix the issue of the Remaining usage as well.

Reply:
yeah i was plugging SDKGasMeter in the test and got weird output